rmf_site_editor/src/widgets/inspector/mod.rs

```diff
@@ -294,10 +282,13 @@ impl<'w, 's> WidgetSystem<Tile> for Inspector<'w, 's> {
         state: &mut SystemState<Self>,
         world: &mut World,
     ) {
+        // TODO(luca) make sure this doesn't show in building preview mode
+        /*
         match world.resource::<State<AppState>>().get() {
             AppState::SiteEditor | AppState::SiteDrawingEditor | AppState::WorkcellEditor => {}
             _ => return,
         }
+        */
```

Comment:
This is the main point that I'm still trying to figure out.
The moment we access `AppState` from any of the widgets, that widget will be necessarily need the site editor and its `AppState`, making it extremely limiting.
If this block was present a downstream app that wanted to use this widget would need to either:
* Add `librmf_site_editor::AppState` to their app and make sure it is in the correct state, regardless of whether they need it or not or
* Copy the whole plugin just to remove this snippet (or customize it for their use case).

The drawback of this specific comment (and the reason it's still marked as a TODO) is that specifically for the site editor this will result in the inspection plugin showing up during the building preview mode, which might not be desirable.
I'm not sure if there is a better solution that doesn't involve a larger refactor.

Reply:
One simple-ish option would be to have an AppState specifically for inspection that we use to toggle the inspector widget on/off explicitly.

I have a feeling that to get a satisfying level of granularity, we'll end up having dedicated state types for each different plugin. That wouldn't be a bad thing if we can additionally still have a high-level `AppState` that automatically manages all those different granular states types.

I don't mind punting that to a follow-up PR so we're not blocking this one in the meantime.